File: remove_comments.js
const fs = require("fs");
const path = require("path");

function removeComments(content, ext) {
  if ([".ts", ".tsx", ".js", ".jsx", ".css", ".scss"].includes(ext)) {
    
    
    
    const pattern =
      /(\"(?:\\\\|\\\"|[^\"])*\"|\'(?:\\\\|\\\'|[^\'])*\'|`(?:\\\\|\\`|[^`])*`)|(\/\*[\s\S]*?\*\/|\/\/.*)/g;

    return content.replace(pattern, (match, string, comment) => {
      if (comment) {
        return "";
      }
      return string;
    });
  }
  return content;
}

function processDirectory(directory) {
  const excludedDirs = new Set([
    ".git",
    "node_modules",
    ".next",
    "dist",
    "build",
    "out",
  ]);
  const allowedExtensions = new Set([
    ".ts",
    ".tsx",
    ".js",
    ".jsx",
    ".css",
    ".scss",
  ]);

  function walk(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
      const filepath = path.join(dir, file);
      const stats = fs.statSync(filepath);

      if (stats.isDirectory()) {
        if (!excludedDirs.has(file)) {
          walk(filepath);
        }
      } else {
        const ext = path.extname(file);
        if (allowedExtensions.has(ext)) {
          try {
            const content = fs.readFileSync(filepath, "utf8");
            const newContent = removeComments(content, ext);

            if (newContent !== content) {
              fs.writeFileSync(filepath, newContent, "utf8");
              console.log(`Processed: ${filepath}`);
            }
          } catch (err) {
            console.error(`Error processing ${filepath}: ${err.message}`);
          }
        }
      }
    }
  }

  walk(directory);
}

const targetDir = "c:\\Users\\Marvan\\Desktop\\BePositive.az";
processDirectory(targetDir);

File: remove_comments.py
import os
import re

def remove_comments(content, ext):
    if ext in ['.ts', '.tsx', '.js', '.jsx', '.css', '.scss']:
        # Regex to match strings and comments
        # Group 1: Strings (to be preserved)
        # Group 2: Comments (to be removed)
        pattern = r'(\"(?:\\\\|\\\"|[^\"])*\"|\'(?:\\\\|\\\'|[^\'])*\'|`(?:\\\\|\\`|[^`])*`)|(/\*[\s\S]*?\*/|//.*)'
        
        def replacer(match):
            if match.group(2): # If it's a comment
                return ""
            else: # If it's a string
                return match.group(1)
        
        return re.sub(pattern, replacer, content)
    return content

def process_directory(directory):
    excluded_dirs = {'.git', 'node_modules', '.next', 'dist', 'build', 'out'}
    allowed_extensions = {'.ts', '.tsx', '.js', '.jsx', '.css', '.scss'}
    
    for root, dirs, files in os.walk(directory):
        dirs[:] = [d for d in dirs if d not in excluded_dirs]
        
        for file in files:
            ext = os.path.splitext(file)[1]
            if ext in allowed_extensions:
                filepath = os.path.join(root, file)
                try:
                    with open(filepath, 'r', encoding='utf-8') as f:
                        content = f.read()
                    
                    new_content = remove_comments(content, ext)
                    
                    if new_content != content:
                        with open(filepath, 'w', encoding='utf-8') as f:
                            f.write(new_content)
                        print(f"Processed: {filepath}")
                except Exception as e:
                    print(f"Error processing {filepath}: {e}")

if __name__ == "__main__":
    process_directory('c:\\Users\\Marvan\\Desktop\\BePositive.az')
